modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/ReconnectionUtils.java

```diff
@@ -134,21 +136,33 @@ public static InterceptorChain createReconnectionInterceptorsChain(ExtensionMode
     InterceptorChain.Builder chainBuilder = InterceptorChain.builder();
 
     if (componentModel instanceof ConnectableComponentModel) {
-      if (((ConnectableComponentModel) componentModel).requiresConnection()) {
-        addConnectionInterceptors(chainBuilder, extensionModel, componentModel, connectionSupplier, reflectionCache);
+      if (requiresConnectionInterceptors(extensionModel, (ConnectableComponentModel) componentModel)) {
```

Comment:
not strictly related to your change, but in the spirit of the boyscout rule let's take the opportunity to refactor this. This class is for reconnection, but The CursorReset is the only interceptor that is actually about reconnecting. This method should receive the chainBuilder and only add the reconnection interceptors only. The ConnectionInterceptor should be added someplace else.

Reply:
Done.
I created an `InterceptorChainUtils` with a `createConnectionInterceptorsChain` which adds the `ConnectionInterceptor` and then delegates to the `ReconnectionUtils` class to add the `CursorResetInterceptor` (if necessary) to the builder as you described.